.github/workflows/test.yaml

```diff
@@ -416,7 +430,7 @@ jobs:
       uses: actions/cache@v3
       with:
         path: "${{ github.workspace }}/.test-results"
-        key: "${{ github.job }}-results-${{ github.run_id }}-${{ matrix.system }}-${{ steps.get-previous-attempt.outputs.previous_attempt }}"
+        key: "${{ github.job }}-results-${{ github.run_id }}-${{ matrix.group }}-${{ steps.get-previous-attempt.outputs.previous_attempt }}"
```

Comment:
what's the effect of caching this way?

Reply:
we use this to re-execute the failed ones by group